### PR TITLE
[Kapacitor] Add way to set kapacitor strategy type

### PR DESCRIPTION
**Pull request by iblackman**
Add possibility to specify `kapacitor` deployment strategy type.

## Why?
It isn't possible to alter the kapacitor deployment strategy. In our case we don't have a RWX PVC and that causes trouble on rollouts of new versions.

---

- [ ] CHANGELOG.md updated (I can't find this file)
- [ ] Rebased/mergable
- [ ] Tests pass
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by iblackman** [comment]
@timhallinflux @russorat @sumo-drosiek I you could check this PR, it is pretty simple. Thanks in advance.

**Comment by alespour** [comment]
@iblackman Could you please try to re-trigger checks?

**Review by timhallinflux — DISMISSED** [review]

**Review by alespour — DISMISSED** [review]

**Comment by timhallinflux** [comment]
@iblackman you will need to sign the CLA for this to be merged. https://www.influxdata.com/legal/cla/

**Comment by iblackman** [comment]
> @iblackman you will need to sign the CLA for this to be merged. https://www.influxdata.com/legal/cla/

@timhallinflux CLA signed.

**Comment by iblackman** [comment]
> @iblackman Could you please try to re-trigger checks?


@alespour  I've push force the last commit to re-trigger the checks

File: charts/kapacitor/templates/deployment.yaml [hunk 1]
```diff
@@ -12,6 +12,8 @@ metadata:
     app: {{ template "kapacitor.fullname" . }}
 spec:
   replicas: 1
+  strategy:
+{{ toYaml .Values.strategy | trim | indent 4 }}
```
Comment (alespour) [inline charts/kapacitor/templates/deployment.yaml:16]: With `strategy` commented out in `values.yaml` by default, the template output is 
```
  strategy:
    null
```
I guess it should be change to
```suggestion
  {{- if .Values.strategy }}
  strategy:
{{ toYaml .Values.strategy | trim | indent 4 }}
  {{- end }}
```

**Review by alespour — CHANGES_REQUESTED** [review]
It's only a small change needed to handle the case where `strategy` is not set.
Thank you for the contribution.

File: charts/kapacitor/templates/deployment.yaml (hunk 1 above)
Comment (timhallinflux) [reply charts/kapacitor/templates/deployment.yaml:16]: @iblackman can you address this? then we are ready to merge

**Comment by timhallinflux** [comment]
@iblackman can you check that last comment?

**Comment by iblackman** [comment]
@alespour @timhallinflux Thanks for the comment, I've added the requested change.

**Review by alespour — APPROVED** [review]

**Review by timhallinflux — APPROVED** [review]

**Pull request merged**